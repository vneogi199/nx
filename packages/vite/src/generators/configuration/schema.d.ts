export interface ViteConfigurationGeneratorSchema {
  uiFramework: 'react' | 'none';
  compiler?: 'babel' | 'swc';
  project: string;
  newProject?: boolean;
  includeVitest?: boolean;
  inSourceTests?: boolean;
  includeLib?: boolean;
  buildTarget?: string;
  serveTarget?: string;
  testTarget?: string;
  skipFormat?: boolean;
  testEnvironment?: 'node' | 'jsdom' | 'happy-dom' | 'edge-runtime' | string;
}
