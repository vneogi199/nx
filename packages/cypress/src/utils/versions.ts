export const nxVersion = require('../../package.json').version;
export const eslintPluginCypressVersion = '^2.13.4';
export const typesNodeVersion = '16.11.7';
export const cypressViteDevServerVersion = '^2.2.1';
export const cypressVersion = '^13.0.0';
export const cypressWebpackVersion = '^2.0.0';
export const webpackHttpPluginVersion = '^5.5.0';
export const viteVersion = '~4.3.9';
export const htmlWebpackPluginVersion = '^5.5.0';
