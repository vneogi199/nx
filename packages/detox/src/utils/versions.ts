export const nxVersion = require('../../package.json').version;

export const detoxVersion = '^20.11.1';
export const testingLibraryJestDom = '5.16.5';
export const configPluginsDetoxVersion = '~6.0.0'; // only required for expo
