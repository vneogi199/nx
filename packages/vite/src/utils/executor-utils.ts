import { printDiagnostics, runTypeCheck } from '@nx/js';
import { join, resolve } from 'path';
import { ViteBuildExecutorOptions } from '../executors/build/schema';
import { ExecutorContext } from '@nx/devkit';
import { ViteDevServerExecutorOptions } from '../executors/dev-server/schema';
import {
  calculateProjectDependencies,
  createTmpTsConfig,
} from '@nx/js/src/utils/buildable-libs-utils';
import { registerTsConfigPaths } from '@nx/js/src/internal';

export async function validateTypes(opts: {
  workspaceRoot: string;
  projectRoot: string;
  tsconfig: string;
}): Promise<void> {
  const result = await runTypeCheck({
    workspaceRoot: opts.workspaceRoot,
    tsConfigPath: join(opts.workspaceRoot, opts.tsconfig),
    mode: 'noEmit',
  });

  await printDiagnostics(result.errors, result.warnings);

  if (result.errors.length > 0) {
    throw new Error('Found type errors. See above.');
  }
}

export function registerPaths(
  projectRoot: string,
  options: ViteBuildExecutorOptions | ViteDevServerExecutorOptions,
  context: ExecutorContext
) {
  const tsConfig = resolve(projectRoot, 'tsconfig.json');
  options.buildLibsFromSource ??= true;

  if (!options.buildLibsFromSource) {
    const { dependencies } = calculateProjectDependencies(
      context.projectGraph,
      context.root,
      context.projectName,
      context.targetName,
      context.configurationName
    );
    const tmpTsConfig = createTmpTsConfig(
      tsConfig,
      context.root,
      projectRoot,
      dependencies
    );

    registerTsConfigPaths(tmpTsConfig);
  } else {
    registerTsConfigPaths(tsConfig);
  }
}
