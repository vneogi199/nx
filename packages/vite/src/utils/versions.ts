export const nxVersion = require('../../package.json').version;
export const viteVersion = '^4.3.9';
export const vitestVersion = '^0.32.0';
export const vitestUiVersion = '^0.32.0';
export const vitePluginReactVersion = '^4.0.0';
export const vitePluginReactSwcVersion = '^3.3.2';
export const viteTsConfigPathsVersion = '^4.2.0';
export const jsdomVersion = '~22.1.0';
export const vitePluginDtsVersion = '~2.3.0';
export const happyDomVersion = '~9.20.3';
export const edgeRuntimeVmVersion = '~3.0.2';

// Coverage providers
export const vitestCoverageC8Version = '^0.32.0';
export const vitestCoverageIstanbulVersion = '^0.32.0';
