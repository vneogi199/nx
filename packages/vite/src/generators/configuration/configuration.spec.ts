import {
  addDependenciesToPackageJson,
  addProjectConfiguration,
  readJson,
  Tree,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { nxVersion } from '../../utils/versions';

import { viteConfigurationGenerator } from './configuration';
import {
  mockAngularAppGenerator,
  mockReactAppGenerator,
  mockReactLibNonBuildableJestTestRunnerGenerator,
  mockReactLibNonBuildableVitestRunnerGenerator,
  mockReactMixedAppGenerator,
  mockUnknownAppGenerator,
  mockWebAppGenerator,
} from '../../utils/test-utils';

describe('@nx/vite:configuration', () => {
  let tree: Tree;

  describe('transform React app to use Vite', () => {
    beforeAll(async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
      mockReactAppGenerator(tree);
      const existing = 'existing';
      const existingVersion = '1.0.0';
      addDependenciesToPackageJson(
        tree,
        { '@nx/vite': nxVersion, [existing]: existingVersion },
        { [existing]: existingVersion }
      );
      await viteConfigurationGenerator(tree, {
        uiFramework: 'react',
        project: 'my-test-react-app',
      });
    });

    it('should add vite packages and react-related dependencies for vite', async () => {
      const packageJson = readJson(tree, '/package.json');
      expect(packageJson.devDependencies).toMatchObject({
        vite: expect.any(String),
        '@vitejs/plugin-react': expect.any(String),
      });
    });

    it('should move index.html to the root of the project', () => {
      expect(tree.exists('apps/my-test-react-app/src/index.html')).toBeFalsy();
      expect(tree.exists('apps/my-test-react-app/index.html')).toBeTruthy();
      expect(
        tree.read('apps/my-test-react-app/index.html', 'utf-8')
      ).toMatchSnapshot();
    });

    it('should create correct tsconfig compilerOptions', () => {
      const tsconfigJson = readJson(
        tree,
        'apps/my-test-react-app/tsconfig.json'
      );
      expect(tsconfigJson.compilerOptions.types).toMatchObject(['vite/client']);
    });

    it('should create vite.config file at the root of the app', () => {
      expect(tree.exists('apps/my-test-react-app/vite.config.ts')).toBe(true);
      expect(
        tree.read('apps/my-test-react-app/vite.config.ts', 'utf-8')
      ).toMatchSnapshot();
    });

    it('should transform workspace.json project config', () => {
      expect(tree.read('workspace.json', 'utf-8')).toMatchSnapshot();
    });
  });

  describe('transform Web app to use Vite', () => {
    beforeAll(async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
      mockWebAppGenerator(tree);
      const existing = 'existing';
      const existingVersion = '1.0.0';
      addDependenciesToPackageJson(
        tree,
        { '@nx/vite': nxVersion, [existing]: existingVersion },
        { [existing]: existingVersion }
      );
      await viteConfigurationGenerator(tree, {
        uiFramework: 'none',
        project: 'my-test-web-app',
      });
    });
    it('should add vite dependencies for vite', async () => {
      const packageJson = readJson(tree, '/package.json');
      expect(packageJson.devDependencies).toMatchObject({
        vite: expect.any(String),
        'vite-tsconfig-paths': expect.any(String),
      });
    });

    it('should move index.html to the root of the project', () => {
      expect(tree.exists('apps/my-test-web-app/src/index.html')).toBeFalsy();
      expect(tree.exists('apps/my-test-web-app/index.html')).toBeTruthy();
      expect(
        tree.read('apps/my-test-web-app/index.html', 'utf-8')
      ).toMatchSnapshot();
    });

    it('should create correct tsconfig compilerOptions', () => {
      const tsconfigJson = readJson(tree, 'apps/my-test-web-app/tsconfig.json');
      expect(tsconfigJson.compilerOptions.types).toMatchObject(['vite/client']);
    });

    it('should create vite.config file at the root of the app', () => {
      expect(tree.exists('apps/my-test-web-app/vite.config.ts')).toBe(true);
      expect(
        tree.read('apps/my-test-web-app/vite.config.ts', 'utf-8')
      ).toMatchSnapshot();
    });

    it('should transform workspace.json project config', () => {
      expect(tree.read('workspace.json', 'utf-8')).toMatchSnapshot();
    });
  });

  describe('do not transform Angular app to use Vite', () => {
    beforeAll(async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
      mockAngularAppGenerator(tree);
    });
    it('should throw when trying to convert', async () => {
      expect.assertions(2);

      try {
        await viteConfigurationGenerator(tree, {
          uiFramework: 'none',
          project: 'my-test-angular-app',
        });
      } catch (e) {
        expect(e).toBeDefined();
        expect(e.toString()).toContain(
          'The project my-test-angular-app cannot be converted to use the @nx/vite executors'
        );
      }
    });
  });

  describe('inform user of unknown targets when converting', () => {
    beforeAll(async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
      mockUnknownAppGenerator(tree);
    });

    it('should throw when trying to convert something unknown', async () => {
      const { Confirm } = require('enquirer');
      const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
      confirmSpy.mockResolvedValue(true);
      expect.assertions(2);

      try {
        await viteConfigurationGenerator(tree, {
          uiFramework: 'none',
          project: 'my-test-random-app',
        });
      } catch (e) {
        expect(e).toBeDefined();
        expect(e.toString()).toContain(
          'Error: Cannot find apps/my-test-random-app/tsconfig.json'
        );
      }
    });

    it('should throw when trying to convert something unknown and user denies conversion', async () => {
      const { Confirm } = require('enquirer');
      const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
      confirmSpy.mockResolvedValue(false);

      expect.assertions(2);

      try {
        await viteConfigurationGenerator(tree, {
          uiFramework: 'none',
          project: 'my-test-random-app',
        });
      } catch (e) {
        expect(e).toBeDefined();
        expect(e.toString()).toContain(
          'Nx could not verify that the executors you are using can be converted to the @nx/vite executors.'
        );
      }
    });
  });

  describe('transform React app to use Vite by providing custom targets', () => {
    describe('transform React app if supported executor is provided', () => {
      beforeEach(async () => {
        tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
        mockReactMixedAppGenerator(tree);
        const existing = 'existing';
        const existingVersion = '1.0.0';
        addDependenciesToPackageJson(
          tree,
          { '@nx/vite': nxVersion, [existing]: existingVersion },
          { [existing]: existingVersion }
        );
        await viteConfigurationGenerator(tree, {
          uiFramework: 'react',
          project: 'my-test-mixed-react-app',
          buildTarget: 'valid-build',
        });
      });
      it('should add vite packages and react-related dependencies for vite', async () => {
        const packageJson = readJson(tree, '/package.json');
        expect(packageJson.devDependencies).toMatchObject({
          vite: expect.any(String),
          '@vitejs/plugin-react': expect.any(String),
        });
      });

      it('should create vite.config file at the root of the app', () => {
        expect(tree.exists('apps/my-test-mixed-react-app/vite.config.ts')).toBe(
          true
        );
      });

      it('should transform workspace.json project config', () => {
        expect(tree.read('workspace.json', 'utf-8')).toMatchSnapshot();
      });
    });

    describe('do NOT transform React app if unsupported executor is provided', () => {
      beforeEach(async () => {
        tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
        mockReactMixedAppGenerator(tree);
        const existing = 'existing';
        const existingVersion = '1.0.0';
        addDependenciesToPackageJson(
          tree,
          { '@nx/vite': nxVersion, [existing]: existingVersion },
          { [existing]: existingVersion }
        );
      });
      it('should throw when trying to convert and user denies', async () => {
        const { Confirm } = require('enquirer');
        const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
        confirmSpy.mockResolvedValue(false);
        expect.assertions(2);

        try {
          await viteConfigurationGenerator(tree, {
            uiFramework: 'none',
            project: 'my-test-mixed-react-app',
            buildTarget: 'invalid-build',
          });
        } catch (e) {
          expect(e).toBeDefined();
          expect(e.toString()).toContain(
            'The build target invalid-build cannot be converted to use the @nx/vite:build executor'
          );
        }
      });

      it('should NOT throw error when trying to convert and user confirms', async () => {
        const { Confirm } = require('enquirer');
        const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
        confirmSpy.mockResolvedValue(true);
        expect.assertions(1);

        try {
          await viteConfigurationGenerator(tree, {
            uiFramework: 'none',
            project: 'my-test-mixed-react-app',
            buildTarget: 'invalid-build',
          });
          expect(
            tree.exists('apps/my-test-mixed-react-app/vite.config.ts')
          ).toBe(true);
        } catch (e) {
          throw new Error('Should not throw error');
        }
      });
    });
  });

  describe('vitest', () => {
    beforeAll(async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
      await mockReactAppGenerator(tree);
      const existing = 'existing';
      const existingVersion = '1.0.0';
      addDependenciesToPackageJson(
        tree,
        { '@nx/vite': nxVersion, [existing]: existingVersion },
        { [existing]: existingVersion }
      );
      await viteConfigurationGenerator(tree, {
        uiFramework: 'react',
        project: 'my-test-react-app',
        includeVitest: true,
      });
    });
    it('should create a vitest configuration if "includeVitest" is true', () => {
      const viteConfig = tree
        .read('apps/my-test-react-app/vite.config.ts')
        .toString();
      expect(viteConfig).toContain('test');
      expect(
        tree.read('apps/my-test-react-app/vite.config.ts', 'utf-8')
      ).toMatchSnapshot();
    });
  });

  describe('library mode', () => {
    beforeEach(async () => {
      tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    });

    it('should add config for building library', async () => {
      addProjectConfiguration(tree, 'my-lib', {
        root: 'my-lib',
      });
      await viteConfigurationGenerator(tree, {
        uiFramework: 'react',
        includeLib: true,
        project: 'my-lib',
        newProject: true,
      });

      const viteConfig = tree.read('my-lib/vite.config.ts').toString();

      expect(viteConfig).toMatch('build: {');
      expect(viteConfig).toMatch("external: ['react'");
      expect(tree.read('my-lib/vite.config.ts', 'utf-8')).toMatchSnapshot();
    });

    it('should set up non buildable library correctly', async () => {
      mockReactLibNonBuildableJestTestRunnerGenerator(tree);
      await viteConfigurationGenerator(tree, {
        uiFramework: 'react',
        project: 'react-lib-nonb-jest',
        includeVitest: true,
      });
      expect(
        tree.read('libs/react-lib-nonb-jest/vite.config.ts', 'utf-8')
      ).toMatchSnapshot();

      expect(tree.read('workspace.json', 'utf-8')).toMatchSnapshot();
    });

    it('should set up non buildable library which already has vite.config.ts correctly', async () => {
      const { Confirm } = require('enquirer');
      const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
      confirmSpy.mockResolvedValue(true);
      expect.assertions(2);

      mockReactLibNonBuildableVitestRunnerGenerator(tree);

      try {
        await viteConfigurationGenerator(tree, {
          uiFramework: 'react',
          project: 'react-lib-nonb-vitest',
          includeVitest: true,
        });
        expect(
          tree.read('libs/react-lib-nonb-vitest/vite.config.ts', 'utf-8')
        ).toMatchSnapshot();

        expect(tree.read('workspace.json', 'utf-8')).toMatchSnapshot();
      } catch (e) {
        throw new Error('Should not throw error');
      }
    });
  });
});
