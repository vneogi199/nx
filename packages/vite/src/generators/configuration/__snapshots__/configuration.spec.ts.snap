// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nx/vite:configuration library mode should add config for building library 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';
import dts from 'vite-plugin-dts';
import * as path from 'path';

export default defineConfig({
  cacheDir: '../node_modules/.vite/my-lib',

  plugins: [
    dts({
      entryRoot: 'src',
      tsConfigFilePath: path.join(__dirname, 'tsconfig.lib.json'),
      skipDiagnostics: true,
    }),
    react(),
    viteTsConfigPaths({
      root: '../',
    }),
  ],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [
  //    viteTsConfigPaths({
  //      root: '../',
  //    }),
  //  ],
  // },

  // Configuration for building your library.
  // See: https://vitejs.dev/guide/build.html#library-mode
  build: {
    lib: {
      // Could also be a dictionary or array of multiple entry points.
      entry: 'src/index.ts',
      name: 'my-lib',
      fileName: 'index',
      // Change this to the formats you want to support.
      // Don't forget to update your package.json as well.
      formats: ['es', 'cjs'],
    },
    rollupOptions: {
      // External packages that should not be bundled into your library.
      external: ['react', 'react-dom', 'react/jsx-runtime'],
    },
  },
});
"
`;

exports[`@nx/vite:configuration library mode should set up non buildable library correctly 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';
import dts from 'vite-plugin-dts';
import * as path from 'path';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/react-lib-nonb-jest',

  plugins: [
    dts({
      entryRoot: 'src',
      tsConfigFilePath: path.join(__dirname, 'tsconfig.lib.json'),
      skipDiagnostics: true,
    }),
    react(),
    viteTsConfigPaths({
      root: '../../',
    }),
  ],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [
  //    viteTsConfigPaths({
  //      root: '../../',
  //    }),
  //  ],
  // },

  // Configuration for building your library.
  // See: https://vitejs.dev/guide/build.html#library-mode
  build: {
    lib: {
      // Could also be a dictionary or array of multiple entry points.
      entry: 'src/index.ts',
      name: 'react-lib-nonb-jest',
      fileName: 'index',
      // Change this to the formats you want to support.
      // Don't forget to update your package.json as well.
      formats: ['es', 'cjs'],
    },
    rollupOptions: {
      // External packages that should not be bundled into your library.
      external: ['react', 'react-dom', 'react/jsx-runtime'],
    },
  },

  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`@nx/vite:configuration library mode should set up non buildable library correctly 2`] = `
"{
  "projects": {
    "react-lib-nonb-jest": {
      "name": "react-lib-nonb-jest",
      "$schema": "../../node_modules/nx/schemas/project-schema.json",
      "root": "libs/react-lib-nonb-jest",
      "sourceRoot": "libs/react-lib-nonb-jest/src",
      "projectType": "library",
      "targets": {
        "lint": {
          "executor": "@nx/linter:eslint",
          "outputs": ["{options.outputFile}"],
          "options": {
            "lintFilePatterns": [
              "libs/react-lib-nonb-jest/**/*.{ts,tsx,js,jsx}"
            ]
          }
        },
        "test": {
          "executor": "@nx/jest:jest",
          "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
          "options": {
            "jestConfig": "libs/react-lib-nonb-jest/jest.config.ts",
            "passWithNoTests": true
          }
        }
      },
      "tags": []
    }
  }
}
"
`;

exports[`@nx/vite:configuration library mode should set up non buildable library which already has vite.config.ts correctly 1`] = `
"import dts from 'vite-plugin-dts';
import * as path from 'path';
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  // Configuration for building your library.
  // See: https://vitejs.dev/guide/build.html#library-mode
  cacheDir: '../../node_modules/.vite/react-lib-nonb-vitest',
  build: {
    lib: {
      // Could also be a dictionary or array of multiple entry points.
      entry: 'src/index.ts',
      name: 'react-lib-nonb-vitest',
      fileName: 'index',
      // Change this to the formats you want to support.
      // Don't forget to update your package.json as well.
      formats: ['es', 'cjs'],
    },
    rollupOptions: {
      // External packages that should not be bundled into your library.
      external: ['react', 'react-dom', 'react/jsx-runtime'],
    },
  },
  plugins: [
    ...[
      react(),
      viteTsConfigPaths({
        root: '../../',
      }),
    ],
    dts({
      entryRoot: 'src',
      tsConfigFilePath: path.join(__dirname, 'tsconfig.lib.json'),
      skipDiagnostics: true,
    }),
  ],

  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`@nx/vite:configuration library mode should set up non buildable library which already has vite.config.ts correctly 2`] = `
"{
  "projects": {
    "react-lib-nonb-vitest": {
      "name": "react-lib-nonb-vitest",
      "$schema": "../../node_modules/nx/schemas/project-schema.json",
      "root": "libs/react-lib-nonb-vitest",
      "sourceRoot": "libs/react-lib-nonb-vitest/src",
      "projectType": "library",
      "targets": {
        "lint": {
          "executor": "@nx/linter:eslint",
          "outputs": ["{options.outputFile}"],
          "options": {
            "lintFilePatterns": [
              "libs/react-lib-nonb-vitest/**/*.{ts,tsx,js,jsx}"
            ]
          }
        },
        "test": {
          "executor": "@nx/vite:test",
          "outputs": ["{projectRoot}/coverage"],
          "options": {
            "passWithNoTests": true,
            "reportsDirectory": "{workspaceRoot}/coverage/{projectRoot}"
          }
        }
      },
      "tags": []
    }
  }
}
"
`;

exports[`@nx/vite:configuration transform React app to use Vite by providing custom targets transform React app if supported executor is provided should transform workspace.json project config 1`] = `
"{
  "projects": {
    "my-test-mixed-react-app": {
      "name": "my-test-mixed-react-app",
      "$schema": "../../node_modules/nx/schemas/project-schema.json",
      "root": "apps/my-test-mixed-react-app",
      "sourceRoot": "apps/my-test-mixed-react-app/src",
      "projectType": "application",
      "targets": {
        "invalid-build": {
          "executor": "@nx/js:tsc",
          "outputs": ["{options.outputPath}"]
        },
        "valid-build": {
          "executor": "@nx/webpack:webpack",
          "outputs": ["{options.outputPath}"]
        },
        "serve": {
          "executor": "@nx/webpack:dev-server",
          "defaultConfiguration": "development",
          "options": {
            "buildTarget": "my-test-mixed-react-app:build",
            "hmr": true
          },
          "configurations": {
            "development": {
              "buildTarget": "my-test-mixed-react-app:build:development"
            },
            "production": {
              "buildTarget": "my-test-mixed-react-app:build:production",
              "hmr": false
            }
          }
        },
        "lint": {
          "executor": "@nx/linter:eslint",
          "outputs": ["{options.outputFile}"],
          "options": {
            "lintFilePatterns": [
              "apps/my-test-mixed-react-app/**/*.{ts,tsx,js,jsx}"
            ]
          }
        },
        "test": {
          "executor": "@nx/jest:jest",
          "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
          "options": {
            "jestConfig": "apps/my-test-mixed-react-app/jest.config.ts",
            "passWithNoTests": true
          }
        }
      },
      "tags": []
    }
  }
}
"
`;

exports[`@nx/vite:configuration transform React app to use Vite should create vite.config file at the root of the app 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/my-test-react-app',

  server: {
    port: 4200,
    host: 'localhost',
  },

  preview: {
    port: 4300,
    host: 'localhost',
  },

  plugins: [
    react(),
    viteTsConfigPaths({
      root: '../../',
    }),
  ],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [
  //    viteTsConfigPaths({
  //      root: '../../',
  //    }),
  //  ],
  // },
});
"
`;

exports[`@nx/vite:configuration transform React app to use Vite should move index.html to the root of the project 1`] = `
"<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>My Test React App</title>
    <base href="/" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="icon" type="image/x-icon" href="favicon.ico" />
  </head>
  <body>
    <div id="root"></div>
    <script type="module" src="/src/main.tsx"></script>
  </body>
</html>
"
`;

exports[`@nx/vite:configuration transform React app to use Vite should transform workspace.json project config 1`] = `
"{
  "projects": {
    "my-test-react-app": {
      "name": "my-test-react-app",
      "$schema": "../../node_modules/nx/schemas/project-schema.json",
      "root": "apps/my-test-react-app",
      "sourceRoot": "apps/my-test-react-app/src",
      "projectType": "application",
      "targets": {
        "build": {
          "executor": "@nx/webpack:webpack",
          "outputs": ["{options.outputPath}"],
          "defaultConfiguration": "production",
          "options": {
            "compiler": "babel",
            "outputPath": "dist/apps/my-test-react-app",
            "index": "apps/my-test-react-app/src/index.html",
            "baseHref": "/",
            "main": "apps/my-test-react-app/src/main.tsx",
            "polyfills": "apps/my-test-react-app/src/polyfills.ts",
            "tsConfig": "apps/my-test-react-app/tsconfig.app.json",
            "assets": [
              "apps/my-test-react-app/src/favicon.ico",
              "apps/my-test-react-app/src/assets"
            ],
            "styles": ["apps/my-test-react-app/src/styles.css"],
            "scripts": [],
            "webpackConfig": "@nx/react/plugins/webpack"
          },
          "configurations": {
            "development": {
              "extractLicenses": false,
              "optimization": false,
              "sourceMap": true,
              "vendorChunk": true
            },
            "production": {
              "fileReplacements": [
                {
                  "replace": "apps/my-test-react-app/src/environments/environment.ts",
                  "with": "apps/my-test-react-app/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false
            }
          }
        },
        "serve": {
          "executor": "@nx/webpack:dev-server",
          "defaultConfiguration": "development",
          "options": {
            "buildTarget": "my-test-react-app:build",
            "hmr": true
          },
          "configurations": {
            "development": {
              "buildTarget": "my-test-react-app:build:development"
            },
            "production": {
              "buildTarget": "my-test-react-app:build:production",
              "hmr": false
            }
          }
        },
        "lint": {
          "executor": "@nx/linter:eslint",
          "outputs": ["{options.outputFile}"],
          "options": {
            "lintFilePatterns": ["apps/my-test-react-app/**/*.{ts,tsx,js,jsx}"]
          }
        },
        "test": {
          "executor": "@nx/jest:jest",
          "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
          "options": {
            "jestConfig": "apps/my-test-react-app/jest.config.ts",
            "passWithNoTests": true
          }
        }
      },
      "tags": []
    }
  }
}
"
`;

exports[`@nx/vite:configuration transform Web app to use Vite should create vite.config file at the root of the app 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';

import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/my-test-web-app',

  server: {
    port: 4200,
    host: 'localhost',
  },

  preview: {
    port: 4300,
    host: 'localhost',
  },

  plugins: [
    viteTsConfigPaths({
      root: '../../',
    }),
  ],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [
  //    viteTsConfigPaths({
  //      root: '../../',
  //    }),
  //  ],
  // },
});
"
`;

exports[`@nx/vite:configuration transform Web app to use Vite should move index.html to the root of the project 1`] = `
"<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>WebappPure</title>
    <base href="/" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="icon" type="image/x-icon" href="favicon.ico" />
  </head>
  <body>
    <workspace-root></workspace-root>
    <script type="module" src="/src/main.ts"></script>
  </body>
</html>
"
`;

exports[`@nx/vite:configuration transform Web app to use Vite should transform workspace.json project config 1`] = `
"{
  "projects": {
    "my-test-web-app": {
      "name": "my-test-web-app",
      "$schema": "../../node_modules/nx/schemas/project-schema.json",
      "projectType": "application",
      "root": "apps/my-test-web-app",
      "sourceRoot": "apps/my-test-web-app/src",
      "tags": [],
      "targets": {
        "build": {
          "executor": "@nx/webpack:webpack",
          "outputs": ["{options.outputPath}"],
          "defaultConfiguration": "production",
          "options": {
            "outputPath": "dist/apps/my-test-web-app",
            "compiler": "babel",
            "main": "apps/my-test-web-app/src/main.ts",
            "tsConfig": "apps/my-test-web-app/tsconfig.app.json",
            "assets": [
              "apps/my-test-web-app/src/favicon.ico",
              "apps/my-test-web-app/src/assets"
            ],
            "index": "apps/my-test-web-app/src/index.html",
            "baseHref": "/",
            "polyfills": "apps/my-test-web-app/src/polyfills.ts",
            "styles": ["apps/my-test-web-app/src/styles.css"],
            "scripts": []
          },
          "configurations": {
            "production": {
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "fileReplacements": [
                {
                  "replace": "apps/my-test-web-app/src/environments/environment.ts",
                  "with": "apps/my-test-web-app/src/environments/environment.prod.ts"
                }
              ]
            }
          }
        },
        "serve": {
          "executor": "@nx/webpack:dev-server",
          "options": {
            "buildTarget": "my-test-web-app:build"
          },
          "configurations": {
            "production": {
              "buildTarget": "my-test-web-app:build:production"
            }
          }
        },
        "lint": {
          "executor": "@nx/linter:eslint",
          "outputs": ["{options.outputFile}"],
          "options": {
            "lintFilePatterns": ["apps/my-test-web-app/**/*.ts"]
          }
        },
        "test": {
          "executor": "@nx/jest:jest",
          "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
          "options": {
            "jestConfig": "apps/my-test-web-app/jest.config.ts",
            "passWithNoTests": true
          }
        }
      }
    }
  }
}
"
`;

exports[`@nx/vite:configuration vitest should create a vitest configuration if "includeVitest" is true 1`] = `
"/// <reference types="vitest" />
import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsConfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/my-test-react-app',

  server: {
    port: 4200,
    host: 'localhost',
  },

  preview: {
    port: 4300,
    host: 'localhost',
  },

  plugins: [
    react(),
    viteTsConfigPaths({
      root: '../../',
    }),
  ],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [
  //    viteTsConfigPaths({
  //      root: '../../',
  //    }),
  //  ],
  // },

  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;
